mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -28,9 +39,10 @@ import kotlin.reflect.KProperty1
 abstract class BaseMvRxViewModel<S : MvRxState>(
     initialState: S,
     debugMode: Boolean,
-    private val stateStore: MvRxStateStore<S> = RealMvRxStateStore(initialState)
+    stateStoreOveride: MvRxStateStore<S>? = null
```

Comment:
@elihart I can make the scope injectable but it should be made clear that the ViewModel owns the scope and it will be cancelled in onClear so you shouldn't pass in another scope like a Fragment's scope

Reply:
@gpeal that sounds good, what do you think @BenSchwab ?

This is what we do
```
    private val customCoroutineContext: CoroutineContext = inject(MvRxDagger.AppGraph::viewModelCoroutineContext).value
) : CoroutineScope by CoroutineScope(customCoroutineContext + SupervisorJob()), BaseMvRxViewModel<S>(
```